pymc3/distributions/bound.py

```diff
@@ -285,25 +121,39 @@ def __call__(self, name, *args, **kwargs):
                 "with the cumulative probability function."
             )
 
-        transform = kwargs.pop("transform", "infer")
-        if issubclass(self.distribution, Continuous):
-            return _ContinuousBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
-        elif issubclass(self.distribution, Discrete):
-            return _DiscreteBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
-        else:
-            raise ValueError("Distribution is neither continuous nor discrete.")
+        initval = kwargs.pop("initval", None)
 
-    def dist(self, *args, **kwargs):
-        if issubclass(self.distribution, Continuous):
-            return _ContinuousBounded.dist(
-                self.distribution, self.lower, self.upper, *args, **kwargs
-            )
+        if initval is None:
+            if lower is not None and upper is not None:
+                initval = np.full(size, floatX((lower + upper) / 2))
+            elif lower is not None:
+                initval = np.full(size, floatX(lower))
+            elif upper is not None:
+                initval = np.full(size, floatX(upper))
+            else:
+                initval = np.full(size, floatX(0))
+
+        lower = lower if lower else -np.inf
+        upper = upper if upper else np.inf
+        res = super().dist([distribution, lower, upper], size=size, **kwargs)
+        res.tag.test_value = initval
+
+        return res
+
+    def logp(value, distribution, lower, upper):
```

Comment:
Not sure why but the value of `value` changes during the `logp` calculation hence the test failures.